blockchain/ethereum.go

```diff
@@ -820,6 +822,55 @@ func (e *EthereumClient) GetLatestFinalizedBlockHeader(ctx context.Context) (*ty
 	return e.Client.HeaderByNumber(ctx, big.NewInt(int64(finalizedBlockNumber)))
 }
 
+// EstimatedFinalizationTime returns the estimated time it takes for a block to be finalized
+// for networks with finality tag enabled, it returns the time between the current and next finalized block
+// for networks with finality depth enabled, it returns the time to mine blocks equal to finality depth
+func (e *EthereumClient) EstimatedFinalizationTime(ctx context.Context) (time.Duration, error) {
+	log.Info().Msg("Calculating estimated finalization time")
+	if e.NetworkConfig.FinalityTag {
+		currentFinalizedHeader, err := e.GetLatestFinalizedBlockHeader(ctx)
+		if err != nil {
+			return 0, err
+		}
+		now := time.Now()
+		// wait for the next finalized block
+		for {
+			time.Sleep(30 * time.Second)
```

Comment:
Seems like you could potentially miss a lot of other finalized headers here. Is it possible to do a historic lookback, instead of just waiting for new blocks as we run the test?

Reply:
yeah. There is a slight chance that I miss a finalized block if it's a super fast chain. I tried to look for last to last finalized block or have a subscriber which can do something like newFinalizedHead (just like newHead) but didn't find any :(